Use brms-bpmsuite-platfrom-bom + remove usage of parent for individual quickstarts

psiroky opened the pull request:
@ryanzhang please take a look at these changes and let me know what you think.

Commit:
Replace integration-platform-bom with brms-bpmsuite-platform-bom

Commit:
[BZ-1225698] don't specify parent pom for individual quickstarts

 * see https://github.com/jboss-developer/jboss-developer-shared-resources/blob/master/guides/CONTRIBUTING.md
   for JDF guidelines which require the individual quickstarts
   to not depend on parent pom

 * versions for the 3rd party artifacts are mostly taken
   from brms-bpmsuite-platfrom-bom

 * this brings certain duplication into the POMs. However, it
   seems to be the lesser of two evils. Using parent brings
   additional issues fro users/customers when they only use/import
   single quickstart and maven/IDE complains about missing. We
   should basically make it as easy as we can for people to use
   these quickstarts -- even if it means slight inconvenience
   on our side

ryanzhang commented on pom.xml:
```diff
@@ -45,15 +45,12 @@
 
-        <!-- Define the version of the JBoss BOMs we want to import to specify tested stacks. -->
+        <!-- Define the version of the JBoss EAP BOM we want to import to specify tested stacks. -->
         <version.org.jboss.bom.eap>6.4.2.GA</version.org.jboss.bom.eap>
```
The version.org.jboss.bom.eap is for defining the jboss-javaee-6.0-with-tools.
If jboss-javaee-6.0-with-tools is really not necessary (I see it is removed), maybe this properties is also redundant?

psiroky replied:
Good catch. I will remove rest of the version properties as well, because they are not needed in the top-level pom.

ryanzhang commented on stateful-ksession/pom.xml:
```diff
@@ -36,6 +33,14 @@
 
+    <properties>
+        <version.org.jboss.bom.brms>6.2.0-SNAPSHOT</version.org.jboss.bom.brms>
+        <version.org.jboss.bom.eap>6.4.2.GA</version.org.jboss.bom.eap>
```
Should we update this one  to 6.4.3.GA since CR28 is align to 6.4.3?
I am going to do ER2 base on 6.4.3 eap dependencies.

psiroky replied:
We should use 6.4.3.GA, but let's do that in different PR/commit.

Commit:
Remove no longer needed version properties

 * these properties are needed in some quickstarts,
   but since they no longer inherit from a parent,
   it does not make sense to have this properties in
   the top-level pom (they need be to directlyin the
   quicstart poms)

ryanzhang commented:
Thanks Petr!
The changes looks good to me. Just commented two very small issue , please double check.

psiroky commented:
I removed the version properties from the top-level pom as they are not needed there.

(I did that in different commit to easily see only the changes. Once merging the commits should be squashed into single one)

psiroky commented:
Merged into 6.2.x
